pkg/workflows/wasm/sdk/runner_wasip1.go

```diff
@@ -18,11 +21,25 @@ var (
 //go:wasmimport env sendResponse
 func sendResponse(respptr unsafe.Pointer, respptrlen int32)
 
+//go:wasmimport env log
+func log(respptr unsafe.Pointer, respptrlen int32)
+
 func bufferToPointerLen(buf []byte) (unsafe.Pointer, int32) {
 	return unsafe.Pointer(&buf[0]), int32(len(buf))
 }
 
 func NewRunner() *Runner {
+
+	wws := &WasmWriteSyncer{
+		logFn: func(msg string) {
```

Comment:
@agparadiso You're dividing up the logic between here and the writer implementation below -- this would probably be simpler if you moved all of this logic down to the writer, and just passed in the log function to the constructor here